client-spark/spark-3/src/main/java/org/apache/spark/shuffle/celeborn/SortBasedShuffleWriter.java

```diff
@@ -280,26 +279,25 @@ private void fastWrite0(scala.collection.Iterator iterator) throws IOException {
           }
         }
       }
-      shuffleWriteTimeSum += System.nanoTime() - insertAndPushStartTime;
       tmpRecords[partitionId] += 1;
     }
-    writeMetrics.incWriteTime(shuffleWriteTimeSum);
   }
 
   private void pushAndSwitch() throws IOException {
+    long start = System.nanoTime();
     if (pipelined) {
```

Comment:
compare with before, seems this approach does not count write record into memory/memory allocation time

Reply:
thus I'm curiosity where is the time-saving gain from.

What is the data listed in the PR description represent? e2e or metrics of shuffle write time?

Reply:
checked Spark native implementation, it also does not include writing records into memory/memory allocation time, thus the change LGTM.

Reply:
Hi @pan3793 , when frequently thread dumping, I found ```shuffleWriteTimeSum += System.nanoTime() - insertAndPushStartTime;``` is very often hit, so I suppose System.nanoTime is not free. Since before this PR this is called for each record, the overhead is not small.

Reply:
Also updated description about the test cases.